setup.sh

```diff
@@ -163,5 +168,5 @@ git diff-index --quiet HEAD || git commit -m "Initial cluster configuration"
 git push "${git_remote}" HEAD
 
 log "Installing Kubernetes cluster"
-wksctl apply --git-url="$(git_http_url "$(git_remote_fetchurl "${git_remote}")")" --git-branch="$(git_current_branch)" ${git_deploy_key}
+wksctl apply --git-url="$(git_http_url "$(git_remote_fetchurl "${git_remote}")")" --git-branch="$(git_current_branch)" "${git_deploy_key}"
```

Comment:
Without the double quotes, this will word-split, and the empty string will not count as an argument.

With the quotes, the behavior changes.
You don't want to pass an empty string as an arg.
I'm pretty sure that crashes `wksctl`.

To get around this, you can either:

1. Mark the word-splitting as intentional:
```bash
# shellcheck disable=SC2086
wksctl apply --git-url="$(git_http_url "$(git_remote_fetchurl "${git_remote}")")" --git-branch="$(git_current_branch)" ${git_deploy_key}
```

2. Use a conditional:
```bash
do_wksctl_apply() {
    wksctl apply \
        --git-url="$(git_http_url "$(git_remote_fetchurl "${git_remote}")")" \
        --git-branch="$(git_current_branch)" \
        "$@"
}
if [ "${git_deploy_key}" ]; then \
    do_wksctl_apply "${git_deploy_key}"
else
    do_wksctl_apply
fi
```

3. Build and apply a bash indexed array
```bash
# bash 3.1+
apply_args=()
[ "${git_deploy_key}" ] && apply_args+=("${git_deploy_key}")
wksctl apply \
    --git-url="$(git_http_url "$(git_remote_fetchurl "${git_remote}")")" \
    --git-branch="$(git_current_branch)" \
    "${apply_args[@]}"
```
```bash
# bash 2+
apply_args=()
[ "${git_deploy_key}" ] && apply_args=("${apply_args[@]}" "${git_deploy_key}")
wksctl apply \
    --git-url="$(git_http_url "$(git_remote_fetchurl "${git_remote}")")" \
    --git-branch="$(git_current_branch)" \
    "${apply_args[@]}"
```

Reply:
Thanks. Good catch.

Chose option 3, but initialized `apply_args` with all the flags passed unconditionally to `wksctl apply`